pkg/okteto/context.go

```diff
@@ -343,7 +343,27 @@ func AddOktetoCredentialsToCfg(cfg *clientcmdapi.Config, cred *types.Credential,
 	if !ok {
 		user = clientcmdapi.NewAuthInfo()
 	}
-	user.Token = cred.Token
+
+	hasKubeToken, err := hasKubeTokenCapabily(oktetoURL)
+	if err != nil {
+		return err
+	}
+	if hasKubeToken {
+		user.Token = ""
+		user.Exec = &clientcmdapi.ExecConfig{
+			Command:            "okteto",
+			Args:               []string{"kubetoken"},
+			APIVersion:         "client.authentication.k8s.io/v1",
+			InstallHint:        "Okteto needs to be installed and in your PATH to use this context. Please visit https://www.okteto.com/docs/getting-started/ for more information.",
+			ProvideClusterInfo: true,
+			InteractiveMode:    "IfAvailable",
+		}
+	} else {
+		// fallback for okteto API before client authentication support
+		// TODO: remove once we stop supporting token based authentication
+		user.Token = cred.Token
+		user.Exec = nil
+	}
```

Comment:
Maybe this could be extracted to its own function, not sure if it worths it or not but it will probably help to unit test this, which is the most important part of this PR

Reply:
Everything was already in place for unit testing. It's a bit verbose but it's very well defined as a table test.

Did manage to find an "odd" behavior with the test: the error was handled at the middle of the function, which could lead to a partial modification of the `cfg` object